lib/auth/clt.go

```diff
@@ -2952,37 +2952,37 @@ func (c *Client) DeleteSemaphore(ctx context.Context, filter services.SemaphoreF
 // UpsertKubeService is used by kubernetes services to report their presence
 // to other auth servers in form of hearbeat expiring after ttl period.
 func (c *Client) UpsertKubeService(ctx context.Context, s services.Server) error {
-	data, err := services.GetServerMarshaler().MarshalServer(s)
+	clt, err := c.grpc()
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	args := &upsertServerRawReq{
-		Server: data,
+	server, ok := s.(*services.ServerV2)
+	if !ok {
+		return trace.BadParameter("invalid type %T", server)
```

Comment:
Nit: Not that it would ever happen, but specify the expected type as well? `expected *services.ServerV2, got %T`